src/odemis/gui/model/__init__.py

```diff
@@ -1407,27 +1408,44 @@ def __init__(self, microscope):
         if model.MD_SAMPLE_CENTERS not in md:
             raise KeyError("Stage has no MD_SAMPLE_CENTERS metadata.")
         centers = md[model.MD_SAMPLE_CENTERS]
-        layout = sample_positions_to_layout(centers)
-        for row_idx, row in enumerate(layout):
-            for column_idx, name in enumerate(row):
-                scintillator_number = int(name.split()[-1])
-                layout[row_idx][column_idx] = scintillator_number
         # SAMPLE_SIZES contains the sizes of the scintillators
         if model.MD_SAMPLE_SIZES not in md:
             raise KeyError("Stage has no MD_SAMPLE_SIZES metadata.")
         sample_sizes = md[model.MD_SAMPLE_SIZES]
+        # Handle error cases for sample centers and sizes
+        if set(centers.keys()) != set(sample_sizes.keys()):
```

Comment:
.keys() are equivalent to sets. I think it's fine to just write:
`if centers.keys() != sample_sizes.keys():`